platform/local/dnsmasq.go

```diff
@@ -147,6 +215,98 @@ func newSegment(s byte) (*Segment, error) {
 		return nil, fmt.Errorf("LinkSetUp() failed: %v", err)
 	}
 
+	// we first create an unique virtual ethernet pair in current network namespace accross the multiple kola instances
+	// we use linux network random port attribution to assert the uniqueness of the IP range in order to avoid IP
+	// range clashes
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return nil, fmt.Errorf("unable to listen on random port: %w", err)
+	}
+
+	defer listener.Close()
```

Comment:
the `listener` needs to live on and only be closed when the network namespace is teared down